studygroups/charts.py

```diff
@@ -0,0 +1,694 @@
+    colors = rotate_colors()
+
+    style = Style(
+        font_family="'Open Sans', 'Helvetica', 'Arial', sans-serif",
+        label_font_size=18.0,
+        major_label_font_size=18.0,
+        value_font_size=18.0,
+        value_label_font_size=18.0,
+        tooltip_font_size=18.0,
+        title_font_size=18.0,
+        legend_font_size=18.0,
+        no_data_font_size=18.0,
+        background='transparent',
+        plot_background='#ffffff',
+        foreground='#515665',
+        foreground_strong='#515665',
+        foreground_subtle='#6c757d',
+        opacity='.6',
+        opacity_hover='.9',
+        transition='400ms ease-in',
+        colors=colors)
+
+    return style
+
+def get_typeform_survey_learner_responses(study_group):
+    return study_group.learnersurveyresponse_set.values_list('response', flat=True)
+
+def get_typeform_survey_facilitator_responses(study_group):
+    return study_group.facilitatorsurveyresponse_set.values_list('response', flat=True)
+
+def get_question_field(study_group, question_id):
+    if study_group.learnersurveyresponse_set.count() > 0:
+        survey_str = study_group.learnersurveyresponse_set.first().survey
+        survey_fields = json.loads(survey_str)['fields']
+        return next((field for field in survey_fields if field["id"] == question_id), None)
+
+def get_response_field(response_str, question_id):
+    response = json.loads(response_str)
+    answers = response['answers']
+    return next((answer for answer in answers if answer["field"]["id"] == question_id), None)
+
+def average(total, divisor):
+    if divisor == 0:
+        return 0
+
+    return round(total / divisor, 2)
+
+
+class LearnerGoalsChart():
+    def __init__(self, study_group, **kwargs):
+        self.chart = pygal.HorizontalBar(style=custom_style(), show_legend=False, max_scale=5, order_min=0, x_title="Learners", **kwargs)
+        self.study_group = study_group
+
+    def get_data(self):
+        data = {}
+        for choice in reversed(GOAL_CHOICES):
+            data[choice[0]] = 0
+
+        signup_questions = self.study_group.application_set.values_list('signup_questions', flat=True)
+
+        # check for responses for field id UXwfFPX0On3f in typeform responses
+        # check if the email for those responses already exists in application_set
+
+        for answer_str in signup_questions:
+            answer = json.loads(answer_str)
+            goal = answer.get('goals', None)
+
+            if goal in data:
+                data[goal] += 1
+
+        return data
+
+    def generate(self, **opts):
+        chart_data = self.get_data()
+        labels = chart_data.keys()
+        serie = chart_data.values()
+
+        self.chart.add('Number of learners', serie)
+        self.chart.x_labels = labels
+
+        if opts.get('output', None) == "png":
+            filename = "report-{}-learner-goals-chart.png".format(self.study_group.uuid)
+            target_path = os.path.join('tmp', filename)
```

Comment:
It's also possible to use [tempfile](https://docs.python.org/3.7/library/tempfile.html). I think the tmp directory you've created here won't persist between deploys, so it's not a priority right now.

Reply:
ok i'll leave it for now